packages/map-template/src/components/Map/Map.jsx

```diff
@@ -84,18 +85,53 @@ function Map({ apiKey, gmApiKey, mapboxAccessToken, venues, venueName, onLocatio
 
     /**
      * Set the venue to show on the map.
+     * If locationId property is present, resolve the promise without fitting the venue.
      *
      * @param {object} venue
      * @param {object} mapsIndoorsInstance
      */
     const setVenue = (venue, mapsIndoorsInstance) => {
         window.localStorage.setItem(localStorageKeyForVenue, venue.name);
-        return mapsIndoorsInstance.fitVenue(venue);
+        if (locationId) {
+            return Promise.resolve();
+        } else {
+            return mapsIndoorsInstance.fitVenue(venue);
+        }
     }
 
     /**
-     * Replace the default tile URL style to the incoming tile style.
+     * Handle the tile style changes and the locationId property.
+     * If the locationId property is present, set the correct floor, center and zoom the map.
+     *
+     * @param {object} miInstance
      */
+    const onBuildingChanged = (miInstance) => {
+
+        onTileStyleChanged(miInstance);
+
+        if (locationId && miInstance) {
+            mapsindoors.services.LocationsService.getLocation(locationId).then(location => {
+                if (location) {
+                    // Set the floor to the one that the location belongs to.
+                    const locationFloor = location.properties.floor;
+                    miInstance.setFloor(locationFloor);
+
+                    // Center the map to the location coordinates.
+                    const locationGeometry = location.geometry.type === 'Point' ? location.geometry.coordinates : location.properties.anchor.coordinates;
+                    miInstance.getMapView().setCenter({ lat: locationGeometry[1], lng: locationGeometry[0] });
+
+                    // Set the zoom level in order to see the location closer on the map.
+                    miInstance?.setZoom(21);
```

Comment:
@andreeaceachir142 What if a Solution has zoom 22 enabled? Could we fish that out of the Solution data somehow so we zoom is as far as possible?

Reply:
Another question is what should happen if the location is a very large room. I imagine in that case we don't want to zoom in as far as possible...?

Reply:
As agreed within the team, when setting the locationId property, the map will center and zoom in to level 21 for now. 
We have created a ticket for implementing a fitLocation() method on the SDK, and we will implement that in a future iteration, once that is done.

Reply:
Does this agreement align with the expectations, @markhougaard ?We will postpone fitting to the geometry and instead just zoom in as much as possible.

Reply:
Yes, there is a task for improving it later on. 🙌🏻

Reply:
Then we still need to handle if we should zoom to level 21 or 22. 

Look at the response in the call to `await mapsindoors.services.SolutionsService.getSolution()`. If the `z22` module exists, then we should zoom to level 22.

Reply:
Added support for checking if the solution has `z22` and if yes, set the zoom level to 22, otherwise set it to 21.